Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/intersection.h

```diff
@@ -26,9 +26,769 @@
+ *   `std::pair<std::size_t, std::size_t>`. This OutputIterator will hold the position of the
+ *  elements in their respective range.
+ * \tparam Kernel a model of `Kernel`
+ *
+ * \param polyline1 the first polyline to check for intersections.
+ * \param polyline2 the second polyline to check for intersections.
+ * \param out output iterator to be filled with all pairs of segments that intersect
+ * \param K an instance of `Kernel`
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `pmesh`.
+ *   If this parameter is omitted, an internal property map for
+ *   `CGAL::vertex_point_t` should be available in `TriangleMesh`\cgalParamEnd
+ *    \cgalParamBegin{geom_traits} an instance of a geometric traits class, model of `PMPSelfIntersectionTraits` \cgalParamEnd
+ * \cgalNamedParamsEnd
+ * \return `out`
+ */
+template < class Polyline
+           , class OutputIterator
+           , class Kernel
+           >
+OutputIterator
+compute_polyline_polyline_intersection( const Polyline& polyline1,
+               const Polyline& polyline2,
+               OutputIterator out,
+               const Kernel& K)
+{
+  typedef typename CGAL::Box_intersection_d::Box_with_info_d<double, 3, std::size_t> Box;
+  typedef typename Kernel::Point_3 Point;
+  // make one box per facet
+  std::vector<Box> boxes1;
+  std::vector<Box> boxes2;
+  boxes1.reserve(
+        std::distance( boost::begin(polyline1), boost::end(polyline1) ) - 1
+        );
+
+  boxes2.reserve(
+        std::distance( boost::begin(polyline2), boost::end(polyline2) ) - 1
+        );
+
+  for(std::size_t i =0; i< polyline1.size()-1; ++i)
+  {
+    Point p1 = polyline1[i];
+    Point p2 = polyline1[i+1];
+    boxes1.push_back(Box(p1.bbox() + p2.bbox(), i));
+  }
+
+  for(std::size_t i =0; i< polyline2.size()-1; ++i)
+  {
+    Point p1 = polyline2[i];
+    Point p2 = polyline2[i+1];
+    boxes2.push_back(Box(p1.bbox() + p2.bbox(), i));
+  }
+
+  // generate box pointers
+  std::vector<const Box*> box1_ptr;
+  std::vector<const Box*> box2_ptr;
+  box1_ptr.reserve(boxes1.size());
+  box2_ptr.reserve(boxes2.size());
+
+  BOOST_FOREACH(Box& b, boxes1)
+      box1_ptr.push_back(&b);
+  BOOST_FOREACH(Box& b, boxes2)
+      box2_ptr.push_back(&b);
+
+  // compute intersections filtered out by boxes
+
+  CGAL::internal::Intersect_polylines<Polyline,
+      Kernel,
+      Box,
+      OutputIterator>
+      intersect_polylines(polyline1,
+                          polyline2,
+                          out,
+                          K);
+
+  std::ptrdiff_t cutoff = 2000;
+  CGAL::box_intersection_d(box1_ptr.begin(), box1_ptr.end(),
+                           box2_ptr.begin(), box2_ptr.end(),
+                           intersect_polylines, cutoff);
+  return intersect_polylines.m_iterator;
+}
+
+
+/*
+ * \ingroup PMP_intersection_grp
+ * detects and records intersections between two triangulated surface meshes.
```

Comment:
reports all the pairs of faces intersecting between two triangulated surface meshes